lib/related_articles/database.rb

```diff
@@ -0,0 +1,118 @@
+# TODO: file an issue about weaviate-ruby relying on Object#present?
+class Object; def present?; !!self; end; end
+
+module RelatedArticles
+  module Database
+    def self.prepare
+      client = RelatedArticles.client
+
+      # Delete old index
+      unless "Not Found" == client.schema.get(class_name: 'EngineeringBlogPost')
+        client.schema.delete(class_name: 'EngineeringBlogPost')
+      end
+
+      # Create new index
+      client.schema.create(
+        class_name: "EngineeringBlogPost",
+        description: "An Artsy engineering blog post",
+        module_config: { "text2vec-openai": {vectorizeClassName: false} },
+        properties: [
+          # properties to vectorize
+          {
+            name: "title",
+            dataType: [ "string" ],
+            description: "The title of the post",
+            "moduleConfig": { "text2vec-openai": { "skip": false } }
+          },
+          {
+            name: "body",
+            dataType: [ "string" ],
+            description: "The plaintext body of the post",
+            "moduleConfig": { "text2vec-openai": { "skip": false } }
+          },
```

Comment:
This defines the schema within Weaviate and in particular specifies which fields will be vectorized by default.

Weaviate itself is configured to integrate with OpenAI's Embeddings API so that we don't have to perform the vectorizing/embedding ourselves here. It happens as a byproduct of inserting these `EngineeringBlogPost` documents into Weaviate.